docs/natspec-format.rst

```diff
@@ -0,0 +1,180 @@
+Note: NatSpec currently does NOT apply to public state variables (see
+`solidity#3418 <https://github.com/ethereum/solidity/issues/3418>`__),
+even if they are declared public and therefore do affect ABI. Note:
+The Solidity compiler only interprets tags if they are external or
+public. You are welcome to use similar comments for your internal and
+private functions, but those will not be parsed.
+
+.. code:: solidity
+
+   pragma solidity ^0.5.6;
+
+   /// @title A simulator for trees
+   /// @author Larry A. Gardner
+   /// @notice You can use this contract for only the most basic simulation
+   /// @dev All function calls are currently implemented without side effects
+   contract Tree {
+       /// @author Mary A. Botanist
+       /// @notice Calculate tree age in years, rounded up, for live trees
+       /// @dev The Alexandr N. Tetearing algorithm could increase precision
+       /// @param rings The number of rings from dendrochronological sample
+       /// @return age in years, rounded up for partial years
+       function age(uint256 rings) external pure returns (uint256) {
+           return rings + 1;
+       }
+   }
+
+.. _header-tags:
+
+Tags
+====
+
+All tags are optional. The following table explains the purpose of each
+NatSpec tag and where it may be used. As a special case, if no tags are
+used then the Solidity compiler will interpret a `///` or `/**` comment
+in the same way as if it were tagged with `@notice`.
+
+=========== =============================================================================== =============================
+Tag                                                                                         Context
+=========== =============================================================================== =============================
+``@title``  A title that should describe the contract/interface                             contract, interface
+``@author`` The name of the author                                                          contract, interface, function
+``@notice`` Explain to an end user what this does                                           contract, interface, function
+``@dev``    Explain to a developer any extra details                                        contract, interface, function
+``@param``  Documents a parameter just like in doxygen (must be followed by parameter name) function
+``@return`` Documents the return type of a contract's function                              function
+=========== =============================================================================== =============================
+
+If your function returns a tuple, like ``(int quotient, int remainder)``
+then use multiple ``@return`` statements in the same format as the
+``@param`` statements.
+
+.. _header-dynamic:
+
+Dynamic expressions
+-------------------
+
+In function documentation, you may use dynamic expressions in any tags.
```

Comment:
I think you should provide an example here, without context, "dynamic expressions" can mean anything. Perhaps also rather call them "evaluated expressions" or "expressions to be evaluated" or "templates"?

Reply:
Example and explanation provided in 9cfd68763bc8248140d2eba73b4e58bd40ae6082